lib/src/Base/Stat/CovarianceModelImplementation.cxx

```diff
@@ -704,6 +706,89 @@ CovarianceModelImplementation::Implementation CovarianceModelImplementation::get
   return clone();
 }
 
+/* Drawing method */
+Graph CovarianceModelImplementation::draw(const UnsignedInteger rowIndex,
+					  const UnsignedInteger columnIndex,
+					  const NumericalScalar tMin,
+					  const NumericalScalar tMax,
+					  const UnsignedInteger pointNumber,
+					  const Bool asStationary,
+					  const Bool correlationFlag) const
+{
+  if (spatialDimension_ != 1) throw NotDefinedException(HERE) << "Error: can draw covariance models only if spatial dimension=1, here spatial dimension=" << spatialDimension_;
+  if (rowIndex >= dimension_) throw InvalidArgumentException(HERE) << "Error: the given row index must be less than " << dimension_ << ", here rowIndex=" << rowIndex;
+  if (columnIndex >= dimension_) throw InvalidArgumentException(HERE) << "Error: the given column index must be less than " << dimension_ << ", here columnIndex=" << columnIndex;
+  if (pointNumber < 2) throw InvalidArgumentException(HERE) << "Error: cannot draw the model with pointNumber<2, here pointNumber=" << pointNumber;
+  // Check if the model is stationary and if we want to draw it this way
+  if (asStationary && isStationary())
+    {
+      // Here we compute the normalization for the correlation instead of using
+      // the amplitude attribute for models for which it is not given
+      NumericalScalar ratio = 1.0;
+      if (correlationFlag)
+	{
+	  ratio = (*this)(0.0)(rowIndex, columnIndex);
+	  // If ratio == 0, the covariance iz zero everywhere
+	  if (ratio == 0.0) ratio == 1.0;
```

Comment:
ratio is the enominator of the correlation. It is zero if and only if the numerator is also zero for the off-diagonal terms, so setting it to 1.0 is a useful trick to avoid a test in the normalization loop.